include/crm/common/scheduler_internal.h

```diff
@@ -176,6 +176,7 @@ struct pcmk__scheduler_private {
     int fence_timeout_ms;           // Value of stonith-timeout property in ms
     const char *placement_strategy; // Value of placement-strategy property
     GList *actions;                 // All scheduled actions
+    GHashTable *singletons;         // Scheduled non-resource actions
```

Comment:
I considered just dropping this -- it's only used by pe_actions.c:lookup_singleton(), which could just search through actions instead. The question is whether the performance of a hash table is worth it in this case. Probably not. Keeping actions as a sorted list (by action key) could be a good middle ground since the find_actions*() functions would benefit as well. Unfortunately I don't have a machine for performance testing ...

Reply:
Maybe add a TODO comment to evaluate, if we ever have another boomstick or script the bring-up of one.

Also the name no longer makes much sense given the new comment, but no need to change it now.

Reply:
These are still singletons in the sense that there will be only one entry in pcmk_scheduler_t:actions with the given action key. For resource actions, the same key could be used for multiple actions (e.g. probing on all nodes).

Reply:
I didn't write very clearly. What I meant is that the significance of the name, or why these actions are singletons, is less obvious now. Not saying it's inaccurate.